### PR TITLE
NEXT-33741 - Improve product page load performance

### PR DESCRIPTION
**Pull request by ghost**
### 1. Why is this change necessary?

If you visit the detail page of a product it first checks, if there is a variant config.
Currently for this check the complete product with all data + associations is fetched, just to check the variant-listing-config.
These changes use the database connection without using the product repository.
It has an even greater impact on performance outside the database, as the product is no longer unnecessarily hydrated.

Before:
image +
image +
image = 2.21 ms

image

After:
image

image

### 2. What does this change do, exactly?
- Changed the `checkVariantListingConfig` method in `ProductDetailRoute` to use the direct database connection to fetch only necessary data

### 3. Describe each step to reproduce the issue or behaviour.
- Open the detail page of a product

### 4. Please link to the relevant issues (if any).
/

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [X] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [X] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Content/Product/SalesChannel/Detail/ProductDetailRoute.php [hunk 1]
```diff
@@ -120,18 +123,31 @@ private function addFilters(SalesChannelContext $context, Criteria $criteria): v
      */
     private function checkVariantListingConfig(string $productId, SalesChannelContext $context): ?string
     {
-        /** @var SalesChannelProductEntity|null $product */
-        $product = $this->productRepository->search(new Criteria([$productId]), $context)->first();
```
Comment (shyim) [inline src/Core/Content/Product/SalesChannel/Detail/ProductDetailRoute.php:124]: Can you try to use here partial data loading aka

$criteria->addFields(['parentId', 'variantListingConfig')):

🤔

File: tests/unit/Core/Content/Product/SalesChannel/Detail/ProductDetailRouteTest.php [hunk 2]
```diff
@@ -46,6 +47,11 @@ class ProductDetailRouteTest extends TestCase
      */
     private SystemConfigService $systemConfig;
 
+    /**
+     * @var MockObject&Connection
+     */
+    private Connection $connection;
```
Comment (shyim) [inline tests/unit/Core/Content/Product/SalesChannel/Detail/ProductDetailRouteTest.php]:
```suggestion
    private MockObject&Connection $connection;
```

Comment (shyim) [reply tests/unit/Core/Content/Product/SalesChannel/Detail/ProductDetailRouteTest.php]: then you can remove the comment :)

File: tests/unit/Core/Content/Product/SalesChannel/Detail/ProductDetailRouteTest.php [hunk 3]
```diff
@@ -64,6 +70,7 @@ protected function setUp(): void
         $this->idsCollection = new IdsCollection();
         $this->productRepository = $this->createMock(SalesChannelRepository::class);
         $this->systemConfig = $this->createMock(SystemConfigService::class);
+        $this->connection = $this->createMock(Connection::class);
```
Comment (shyim) [inline tests/unit/Core/Content/Product/SalesChannel/Detail/ProductDetailRouteTest.php:72]: was this method not tested? as you didn't had to configure the connection mock 🤔

Comment (akf-bw) [reply tests/unit/Core/Content/Product/SalesChannel/Detail/ProductDetailRouteTest.php:73]: No currently there is no test for this method.

Comment (ghost) [reply tests/unit/Core/Content/Product/SalesChannel/Detail/ProductDetailRouteTest.php:72]: I have added a new test for this method. I don't know if it works properly in all scenarios, so please check it out. 👍

**Comment by shyim** [comment]
Thanks!

**Pull request closed without merging**